extendr-api/src/wrapper/matrix.rs

```diff
@@ -220,9 +229,9 @@ where
         self.as_typed_slice_mut().unwrap()
     }
 
-    /// Get the dimensions for this array.
-    pub fn dim(&self) -> &D {
-        &self.dim
```

Comment:
This is a breaking change. We should keep this method.

Reply:
We have a `get_dim`. I don't know why we had a `dim` before, and who used it. But alternatively, we inject a call to `get_dim` in here.

Reply:
I can't even mimic the previous behavior, as we no longer have with a const array anymore, that is defined by Rust and used by Rust. Rather, we are not using R's `dim` array.

Reply:
Hi @CGMossa,
Since `dim()` is a public function, I think we should keep the original name and output.
So, let's have `dim()` to return the n-dimension. I will revert the change from `ndim()` back to `dim().
At least in the current version.